Assets/_Laz/Scripts/Managers/CheckPointManager.cs

```diff
@@ -1,28 +1,85 @@
-using System.Collections.Generic;
+using PerigonGames;
 using UnityEngine;
 
 namespace Laz
 {
     public class CheckPointManager : MonoBehaviour
     {
-        [SerializeField] private List<Transform> _checkPoints = null;
-        public Transform LastCheckPoint { get; private set; } = null;
+        [SerializeField] private Checkpoint[] _checkpoints;
+        private Checkpoint _activeCheckpoint = null;
+        private Vector3 _initialCheckPointPosition = Vector3.zero;
 
         public void Initialize()
         {
-            UpdateCheckPoint();
+            if (_checkpoints.IsNullOrEmpty())
+            {
+                return;
+            }
+            
+            for (int i = 0; i < _checkpoints.Length; i++)
+            {
+                Checkpoint checkpoint = _checkpoints[i];
+
+                if (checkpoint == null)
+                {
+                    continue;
+                }
+                
+                checkpoint.OnCheckpointActivation += SetNewCheckpoint;
+
+                if (i == 0)
+                {
+                    _activeCheckpoint = checkpoint;
+                    _activeCheckpoint.IsActiveCheckpoint = true;
+                    _initialCheckPointPosition = _activeCheckpoint.transform.position;
+                }
+            }
+        }
+        
+        public Vector3 GetActiveCheckpointPosition()
+        {
+            return _activeCheckpoint != null ? _activeCheckpoint.transform.position : _initialCheckPointPosition;
+        }
+        
+        //Not sure how this will be implemented properly with current setup
+        public void Reset()
+        {
+            if (_checkpoints.IsNullOrEmpty())
+            {
+                return;
+            }
+
+            foreach (Checkpoint checkpoint in _checkpoints)
+            {
+                if (checkpoint != null)
+                {
+                    checkpoint.IsActiveCheckpoint = false;
+                }
+            }
         }
 
-        //TODO - This is subject to change, data structure used can/will change depending on how checkpoints are chosen
-        public void UpdateCheckPoint()
+        private void SetNewCheckpoint(Checkpoint checkpoint)
```

Comment:
do you really need to pass in the whole checkpoint? why not just pass in a Vector3? 
Since all we really need is a position to spawn back to.

Reply:
Shouldn't Laz's SpawnPoint be set here?